packages/dds/tree/src/feature-libraries/detachedFieldIndexSummarizer.ts

```diff
@@ -4,34 +4,100 @@
  */
 
 import { bufferToString } from "@fluid-internal/client-utils";
+import { assert } from "@fluidframework/core-utils/internal";
 import type { IChannelStorageService } from "@fluidframework/datastore-definitions/internal";
 import type {
 	IExperimentalIncrementalSummaryContext,
 	ISummaryTreeWithStats,
 	ITelemetryContext,
+	MinimumVersionForCollab,
 } from "@fluidframework/runtime-definitions/internal";
-import { createSingleBlobSummary } from "@fluidframework/shared-object-base/internal";
+import { SummaryTreeBuilder } from "@fluidframework/runtime-utils/internal";
 
 import type { DetachedFieldIndex } from "../core/index.js";
 import type {
 	Summarizable,
 	SummaryElementParser,
 	SummaryElementStringifier,
 } from "../shared-tree-core/index.js";
-import type { JsonCompatibleReadOnly } from "../util/index.js";
+import {
+	brand,
+	readAndParseSnapshotBlob,
+	type Brand,
+	type JsonCompatibleReadOnly,
+} from "../util/index.js";
+import { FluidClientVersion } from "../codec/index.js";
 
 /**
  * The storage key for the blob in the summary containing schema data
  */
 const detachedFieldIndexBlobKey = "DetachedFieldIndexBlob";
 
+/**
+ * The storage key for the blob containing metadata for the detached field index's summary.
+ */
+export const detachedFieldIndexMetadataKey = ".metadata";
+
+/**
+ * The versions for the detached field index summary.
+ */
+export const DetachedFieldIndexSummaryVersion = {
+	/**
+	 * Version 0 represents summaries before versioning was added. This version is not written.
+	 * It is only used to avoid undefined checks.
+	 */
+	v0: 0,
+	/**
+	 * Version 1 adds metadata to the detached field index summary.
+	 */
+	v1: 1,
+	/**
+	 * The latest version of the detached field index summary. Must be updated when a new version is added.
+	 */
+	vLatest: 1,
+} as const;
+export type DetachedFieldIndexSummaryVersion = Brand<
+	(typeof DetachedFieldIndexSummaryVersion)[keyof typeof DetachedFieldIndexSummaryVersion],
+	"DetachedFieldIndexSummaryVersion"
+>;
+
+/**
+ * The type for the metadata in the detached field index's summary.
+ * Using type definition instead of interface to make this compatible with JsonCompatible.
+ */
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type DetachedFieldIndexSummaryMetadata = {
+	/** The version of the detached field index summary. */
+	readonly version: DetachedFieldIndexSummaryVersion;
+};
+
+/**
+ * Returns the summary version to use as per the given minimum version for collab.
+ */
+function minVersionToDetachedFieldIndexSummaryVersion(
+	version: MinimumVersionForCollab,
+): DetachedFieldIndexSummaryVersion {
+	return version < FluidClientVersion.v2_73
+		? brand(DetachedFieldIndexSummaryVersion.v0)
```

Comment:
the fact we have to brand our own constants to their intended type is removing a lot of the value of having named constants and a branded type. See my other comments about fixing this.